File: frontend/src/api/orderChange.ts
import { request, type ApiPage } from '@/utils/request'
import { asPage } from '@/utils/apiNormalize'

export interface OrderChangeQuery {
  pageNum: number
  pageSize: number
  orderId?: number | string | null
}

export interface OrderChangeItem {
  id: number
  orderId: number
  orderDetailId?: number
  changeType: number
  changeContent: string
  beforeValue?: string
  afterValue?: string
  changeReason: string
  approveStatus: number
  approveRemark?: string
  applyTime?: string
  approveTime?: string
}

export interface OrderChangeCreateDTO {
  orderId: number | string
  orderDetailId?: number
  changeType: number
  changeContent: string
  beforeValue?: string
  afterValue?: string
  changeReason: string
}

export interface OrderChangeApproveDTO {
  approveStatus: number
  approveRemark?: string
}

export const orderChangeApi = {
  page: async (params: OrderChangeQuery) =>
    asPage<OrderChangeItem>(await request.get<ApiPage<OrderChangeItem>, ApiPage<OrderChangeItem>>('/v1/order-changes', { params }), params.pageNum, params.pageSize),
  detail: (id: number | string) => request.get<OrderChangeItem, OrderChangeItem>(`/v1/order-changes/${id}`),
  create: (data: OrderChangeCreateDTO) =>
    request.post<number, number>('/v1/order-changes', data),
  approve: (id: number | string, data: OrderChangeApproveDTO) =>
    request.post<void, void>(`/v1/order-changes/${id}/approve`, data),
}

File: backend/src/main/java/com/supplier/order/entity/OrderChange.java
package com.supplier.order.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.supplier.common.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDateTime;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("order_change")
public class OrderChange extends BaseEntity {

    private String changeNo;
    private Long orderId;
    private Long orderDetailId;
    private Integer changeType;
    private String changeContent;
    private String beforeValue;
    private String afterValue;
    private String changeReason;
    private Long applyBy;
    private LocalDateTime applyTime;
    private Long approveBy;
    private LocalDateTime approveTime;
    private Integer approveStatus;
    private String approveRemark;
}
